libs/wingsdk/src/target-tf-aws/bucket.ts

```diff
@@ -117,6 +124,25 @@ export class Bucket extends cloud.BucketBase {
   private envName(): string {
     return `BUCKET_NAME_${this.node.addr.slice(-8)}`;
   }
+
+  /**
+   * AWS Bucket names must be between 3 and 63 characters.
+   * You can use lowercase alphanumeric characters, dot(.) and dash (-),
+   * must begin and end with alphanumeric character, can not begin
+   * with 'xn--' or ending with '-s3alias' and must not contain
+   * two adjacent dots.
+   */
+  private sanitizeName(name: string): string {
+    const nameProps: NameOptions = {
+      maxLen: 63,
+      case: CaseConventions.LOWERCASE,
+      regexMatch: /(([(.)\1+]+)[^a-z0-9\-]+)/g,
+      charReplacer: "-",
+      avoidStartWith: "xn--",
```

Comment:
I think we can make the API a bit more general by removing this field (`avoidStartsWith`). Instead of throwing an error when a generated name happens to start with "xn--", we can just add it to the `regexMatch` so it gets replaced with "-" anywhere in the string:

```suggestion
      regexMatcher: /(([(.)\1+]+)[^a-z0-9\-]+)|xn--/g,
      regexReplacer: "-"
```

Reply:
the problem is if a node.id was `xn-- somethink` will be replaced to `-somenthink` and bucket names must begin and end with a letter or number

I was thinking in put `forbiddenStartWith` and `forbiddenEndWith`

Reply:
I see... another option might be to do the assertion/throw `bucket.ts` so `ResourceNames` only just responsible for name generation. But I'm ok leaving this in this field if you think there are multiple resources this could be useful for

Reply:
I actually only see these limitations on AWS Buckets

I think validating on `bucket.ts` is a great option